mikrotik/resource_bgp_instance.go

```diff
@@ -120,10 +128,15 @@ func resourceBgpInstanceRead(d *schema.ResourceData, m interface{}) error {
 		return nil
 	}
 
-	return bgpInstanceToData(bgpInstance, d)
+	err = bgpInstanceToData(bgpInstance, d)
```

Comment:
I'd prefer to update ToData functions to return a `diag.Diagnostics` type rather than update each create, update and delete function.

We could do something like the following to reduce the amount of return statements that would need updated.

```
func bgpInstanceToData(b *client.BgpInstance, d *schema.ResourceData) diag.Diagnostics {
   values := map[string]interface{}{
      "name": b.Name
     "client_to_client_reflection": b.ClientToClientReflection
     
     [ ... ]
   }
   d.SetId(b.Name)
   for attrKey, attrValue := range values { 
     if err := d.Set(attrKey, attrValue); err != nil {
        return diag.FromErr(err)
      }
   }
```

Reply:
Fair enough, it sounds like a better way to do it. I didn't focus much on the actual logic inside the code in order not to bloat this particular PR. I want to put some time into a couple more improvements, but I plan to raise them as separate PRs. 🙂